crates/re_selection_panel/src/selection_panel.rs

```diff
@@ -221,7 +225,34 @@ impl SelectionPanel {
             }
 
             Item::DataResult(space_view_id, instance_path) => {
-                blueprint_ui_for_data_result(ctx, blueprint, ui, *space_view_id, instance_path);
+                let Some(space_view) = blueprint.space_view(space_view_id) else {
+                    return;
+                };
+
+                let visualizer_collection = ctx
+                    .space_view_class_registry
+                    .new_visualizer_collection(space_view.class_identifier());
+
+                let Some(view_state) = view_states
+                    .get(*space_view_id)
+                    .map(|states| states.view_state.as_ref())
+                else {
+                    return;
+                };
+
+                let view_ctx = ViewContext {
+                    viewer_ctx: ctx,
+                    view_state,
+                    visualizer_collection: &visualizer_collection,
+                };
```

Comment:
A helper would be nice.

```suggestion
               if let Some(view_ctx) = ctx.as_view_context() else {
                   return;
               };
```

edit: I saw elsewhere that you are constructing it slightly differently, so this may not make much sense

Reply:
I'm going to come back and do this one on top of the next PR in this stack once I know the different variants of how it needs to get built.

Reply:
makes sense 👍🏻